server/executor/assertion_runner.go

```diff
@@ -160,6 +160,23 @@ func (e *defaultAssertionRunner) setResults(result *openapi.TestRunResult, testR
 	result.AssertionResultState = allTestsPassed
 }
 
-func (e *defaultAssertionRunner) RunAssertions(request AssertionRequest) {
-	e.inputChannel <- request
+func (e *defaultAssertionRunner) RunAssertions(ctx context.Context, result openapi.TestRunResult) error {
```

Comment:
This is adding an unnecesary responsibility to this function. I think the controller can easily do this setup and keep the assertion runner just running assertions